src/core/include/shapes.h

```diff
@@ -3,14 +3,18 @@
 
 #include <stdio.h>
 
-#include "common.h"
+#include "../../common/include/common.h"
 
 class BasicShape {
  public:
-  explicit BasicShape(const size_t id) : id_(id) {}
-  BasicShape(const size_t id, const Box3D& bbox) : id_(id), bbox_(bbox) {}
-  size_t id() const { return id_; }
-  const Box3D& bbox() const { return bbox_; }
+	explicit BasicShape(const size_t id) : id_(id) {}
+	BasicShape(const size_t id, const Box3D& bbox,
+	           ShapeType shape_type = ShapeType::Undefined)
+			: id_(id), bbox_(bbox), shape_type_(shape_type) {}
```

Comment:
We shall not put a shape_type as an argument at all, even with default parameter. It should be set directly in constructor demending on the class